tests/rule_test.php

```diff
@@ -0,0 +1,315 @@
+    }
+
+    /**
+     * Verifies access is blocked when a required purpose is hidden.
+     *
+     * @covers ::prevent_access
+     * @covers ::prevent_new_attempt
+     */
+    public function test_prevent_access_blocks_when_required_purpose_hidden(): void {
+        $rule = $this->create_rule(
+            $this->build_handler([
+                'availability' => ['available' => ai_manager_utils::AVAILABILITY_AVAILABLE],
+                'purposes' => [
+                    ['purpose' => 'feedback', 'available' => ai_manager_utils::AVAILABILITY_HIDDEN, 'errormessage' => ''],
+                    ['purpose' => 'translate', 'available' => ai_manager_utils::AVAILABILITY_AVAILABLE],
+                ],
+            ])
+        );
+
+        $expected = get_string('error_aipurposeunavailable', 'quizaccess_ai', 'feedback');
+        $this->assertSame($expected, $rule->prevent_access());
+        $this->assertSame($expected, $rule->prevent_new_attempt(0, null));
+    }
+
+    /**
+     * Verifies access is blocked when a required purpose is disabled.
+     *
+     * @covers ::prevent_access
+     * @covers ::prevent_new_attempt
+     */
+    public function test_prevent_access_blocks_when_required_purpose_disabled(): void {
+        $rule = $this->create_rule(
+            $this->build_handler([
+                'availability' => ['available' => ai_manager_utils::AVAILABILITY_AVAILABLE],
+                'purposes' => [
+                    [
+                        'purpose' => 'feedback',
+                        'available' => ai_manager_utils::AVAILABILITY_DISABLED,
+                        'errormessage' => 'ai_manager->errormessage',
+                    ],
+                    ['purpose' => 'translate', 'available' => ai_manager_utils::AVAILABILITY_AVAILABLE],
+                ],
+            ])
+        );
+
+        $expected = 'ai_manager->errormessage';
+        $this->assertSame($expected, $rule->prevent_access());
+        $this->assertSame($expected, $rule->prevent_new_attempt(0, null));
+    }
+
+    /**
+     * Verifies multiple purpose messages are combined.
+     *
+     * @covers ::prevent_access
+     * @covers ::prevent_new_attempt
+     */
+    public function test_prevent_access_combines_multiple_purpose_messages(): void {
+        $rule = $this->create_rule(
+            $this->build_handler([
+                'availability' => ['available' => ai_manager_utils::AVAILABILITY_AVAILABLE],
+                'purposes' => [
+                    [
+                        'purpose' => 'feedback',
+                        'available' => ai_manager_utils::AVAILABILITY_HIDDEN,
+                        'errormessage' => '',
+                    ],
+                    [
+                        'purpose' => 'translate',
+                        'available' => ai_manager_utils::AVAILABILITY_DISABLED,
+                        'errormessage' => 'translate disabled',
+                    ],
+                ],
```

Comment:
would be nice to have a comment here that you mock the return structure of the ai manager return structure and the error messages are examples, not the real ones